packages/labs/analyzer/src/lib/model.ts

```diff
@@ -316,29 +327,32 @@ export interface VariableDeclarationInit extends DeclarationInit {
 }
 
 export class VariableDeclaration extends Declaration {
-  readonly node:
+  readonly type: Type | undefined;
+  declare readonly node:
```

Comment:
instead, what if we gave Declaration a type parameter with a default value of ts.Node and the subclasses like VariableDeclaration could set it when extending?